### ISSUE TITLE
WebViewControlNewWindowRequestedEventArgs.Uri loses hash/anchor

### ISSUE THREAD

**Issue by KeithHenry** [windows-toolkit/Microsoft.Toolkit.Win32#54]
## I'm submitting a...
- Bug report (I searched for similar issues and did not find one)

## Current behavior
I have a `WebView` control in a VSTO (Outlook) custom task pane. It loads a web page that has relevant links for the user's current mail.

These links include a `#` component that is critical to the path, like `https://example.com/site#resource:id` - that `#resource:id` is essential and is used by client side JS at the target site.

I've subscribed to the `NewWindowRequested` event, but in this `WebViewControlNewWindowRequestedEventArgs.Uri` the `#resource:id` has been stripped off. 

This would be ok server-side as the hash is not sent, but client side it is a bug.

## Expected behavior
If the link in the `WebView` is `<a href="https://example.com/site#resource:id">test</test>` then `WebViewControlNewWindowRequestedEventArgs.Uri.AbsoluteUri` should be `https://example.com/site#resource:id`.

## Minimal reproduction of the problem with instructions
- In Visual Studio 2017 create a VSTO project for Outlook.
- Add a XAML `UserControl`
```xaml
<UserControl ...>
    <Grid>
        <Controls:WebView ... NewWindowRequested="wbCtrl_NewWindowRequested"/>
    </Grid>
</UserControl>
```
- Set the source to a page with `<a href="https://example.com/site#resource:id">test</test>`
- In `wbCtrl_NewWindowRequested` use your favourite logging tool or just add a breakpoint to check the contents of `WebViewControlNewWindowRequestedEventArgs.Uri`.
- Add a wrapper `Forms.UserControl` with and `ElementHost` containing the above control as its child
- In the application `Startup` event use `this.CustomTaskPanes.Add(...` to add the wrapper control. 
- Run the program with Outlook
- Click on the `test` link in the custom task pane.

Note that most of these steps may be uneccessary if a `WebView` in a standalone application exhibits the same symptoms, I haven't tested it in other contexts yet.

## Environment

Nuget Package/Package Version: 
- `Microsoft.Toolkit.Wpf.UI.Controls.WebView` `5.0.0`

Windows 10 Build Number:
- [ ] Fall Creators Update (16299)
- [x] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [ ] Insider Build (build number: )

App min and target version:
- [ ] Fall Creators Update (16299)
- [x] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [ ] Insider Build (xxxxx)

Device form factor:
- [x] Desktop
- [x] Mobile
- [ ] Xbox
- [ ] Surface Hub
- [ ] IoT

Visual Studio 
- [x] 2017 (version: 15.8.9)
- [ ] 2017 Preview (version: )

**Comment by verelpode** [comment]
> most of these steps may be uneccessary if a WebView in a standalone application exhibits the same symptoms

Yes I think the same problem does exist in a standalone application therefore the part about VSTO/Outlook is probably irrelevant/unnecessary as you said.  I can confirm that I'm also experiencing an inability to load webpages in response to the `NewWindowRequested` event (with WPF WebView 6.0.0-preview7).   I need to investigate this problem deeper and collect details.

> according to the docs the event NewWindowRequested doesn't work as expected.

I wish the documentation explained exactly what _"doesn't work as expected"_ means in the case of NewWindowRequested.  Or at least if the docs don't explain it, can someone explain it here in GitHub?   

At first glance, `NewWindowRequested` looks easy to use, but I've not yet succeeded in using it successfully -- I create a new WebView in response to `NewWindowRequested` and invoke `WebView.Navigate(WebViewControlNewWindowRequestedEventArgs.Uri)` but the webpage fails to load.

**Comment by verelpode** [comment]
I investigated further.  My problem was a different problem, not the hash issue.  I discovered that `WebViewControlNewWindowRequestedEventArgs.Uri` can only be loaded in a `WebView` instance that is using the **same** instance of `WebViewControlProcess`.   I suppose this is reasonable, but the documentation did not explain this.  Is this what _"doesn't work as expected"_ means?  Does there exist additional things about `NewWindowRequested` that don't work as expected?

I like the ability to clear session state by invoking `WebViewControlProcess.Terminate()` followed by creating a new process instance, but the **problem** is:  Now `Microsoft.Toolkit.Wpf.UI.Controls.WebView` has more functionality than `Windows.UI.Xaml.Controls.WebView` and this extra functionality is good except that it will **disappear** when we finish migrating our apps to UWP and switch to `Windows.UI.Xaml.Controls.WebView`.   Thus I think `Windows.UI.Xaml.Controls.WebView` should be extended in order to make it no less powerful than `Microsoft.Toolkit.Wpf.UI.Controls.WebView`.

In our environment, 2 things are very important for us:
* Reliable ability to display popup webpages via the `NewWindowRequested` event (without losing important info such as the hashes/anchors).
* Good control over HTTP session state and cookies, in order to ensure that `NewWindowRequested` can be used successfully, and to ensure that each person/user only ever has access to his/her own session state and cookies;  never gains access to another person's session state and cookies.
